src/core/server/saved_objects/import/check_conflict_for_data_source.ts

```diff
@@ -74,12 +92,37 @@ export async function checkConflictsForDataSource({
         /**
          * Only update importIdMap and filtered objects
          */
+
+        // Some visualization types will need special modifications, like Vega visualizations
+        if (object.type === 'visualization') {
+          const vegaSpec = extractVegaSpecFromSavedObject(object);
+
+          if (!!vegaSpec) {
+            const previousDataSourceTitle =
+              previoudDataSourceId && savedObjectsClient
+                ? await getDataSourceTitleFromId(previoudDataSourceId, savedObjectsClient)
```

Comment:
in this logic, previousDataSourceId is something we get from object id which is not the case for vega right, since we don't save the data source id into vega vis object id?

Reply:
Yes. Now since we attach datasource references, we do not need a previous datasource id. When a vega saved object is imported from to a datasource, any local queries will be modified to append a `data_source_name` and the new datasourceid reference attached to the references list.